lib/puppet/provider/elasticsearch_plugin/plugin.rb

```diff
@@ -90,7 +98,12 @@ def destroy
   def es_version
     return @es_version if @es_version
     begin
-      version = es('-v') # ES 1.x
+      os = Facter['osfamily'].value
+      if os == 'OpenBSD'
+        version = pkg_info('elasticsearch').lines.first
```

Comment:
running the plain binary I get:
# /usr/local/elasticsearch/bin/elasticsearch --version
You must set the ES_CLASSPATH var

So I would have to run it like this:
ES_INCLUDE="/etc/elasticsearch/elasticsearch.in.sh" /usr/local/elasticsearch/bin/elasticsearch --version

to get the version output,
so I would have to special case it in any way. 
If you prefer, I can look into setting the environment variable, before attempting to run the command, instead of using pkg_info. Shouldn't be too complicated.